assets/account/src/lib.rs

```diff
@@ -88,5 +101,35 @@ blueprint! {
                 }
             }
         }
+
+        /// Create proof of resource.
+        pub fn create_proof_by_amount(
+            &self,
+            amount: Decimal,
+            resource_def_id: ResourceDefId,
+        ) -> Proof {
+            let vault = self.vaults.get(&resource_def_id);
+            match vault {
+                Some(vault) => vault.create_proof_by_amount(amount),
+                None => {
+                    panic!("No such resource in account");
+                }
+            }
+        }
+
+        /// Create proof of resource.
+        pub fn create_proof_by_ids(
+            &self,
+            ids: BTreeSet<NonFungibleId>,
+            resource_def_id: ResourceDefId,
+        ) -> Proof {
+            let vault = self.vaults.get(&resource_def_id);
```

Comment:
Should we check for ids.is_empty() here? Should also add comment on what to expect if ids.is_empty()

Reply:
Yes